server/controllers/finance/creditorGroups.js

```diff
@@ -10,6 +10,13 @@ const db = require('../../lib/db');
 const NotFound = require('../../lib/errors/NotFound');
 const uuid = require('node-uuid');
 
+// global constants
+const FAILURE_DELETING_CREDITOR_GROUP = {
```

Comment:
:+1:  This is a good idea.

This is how I accomplish the same thing:
```js
promise.then((error) => {
  if (error) {
    throw new BadRequest('Cannot delete the creditor group, an error occurred', 'CREDITOR_GROUP.FAILURE_DELETE');
  }
  // etc ...
});
```

It's almost the same as yours, except that the `status` is in the name ([BadRequest](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.1)) and it allows me to put it in more details about the error.  For example:
```js
promise.then((error) => {
  if (error) {
    throw new BadRequest(`Cannot delete the creditor group with id ${req.params.id}`, 'CREDITOR_GROUP.FAILURE_DELETE');
  }
  // etc ...
});
```

There is nothing wrong with your error, but I think `BadRequest` will be more useful to you.